packages/wrangler/src/__tests__/tail.test.ts

```diff
@@ -733,6 +727,51 @@ describe("tail", () => {
 			expect(std.warn).toMatchInlineSnapshot(`""`);
 		});
 	});
+
+	describe("disconnects", () => {
+		it("errors when the websocket is already closed", async () => {
+			const api = mockWebsocketAPIs();
+			await api.closeHelper();
+
+			await expect(runWrangler("tail test-worker")).rejects.toThrow();
+		});
+
+		it("errors when the websocket stops reacting to pings (pretty format)", async () => {
+			const api = mockWebsocketAPIs();
+			jest.useFakeTimers({
+				doNotFake: ["setTimeout"],
+			});
+			// Block the websocket from replying to the ping
+			jest.spyOn(MockWebSocket2.prototype, "ping").mockImplementation();
+			await runWrangler("tail test-worker --format=pretty");
+			await api.ws.connected;
+			// The ping is sent every 2 secs, so it should not fail until the second ping is due.
+			await jest.advanceTimersByTimeAsync(2000);
+			await expect(
+				jest.advanceTimersByTimeAsync(2000)
```

Comment:
These timers need an update too! Perhaps it could be a shared constant from the actual code & the test code?

Reply:
Yeah I missed them first time around but then the tests failed (as is desired!) and I fixed it in an amend commit.
I don't really want to export the constant just for the tests. The tests themselves should tell us if this constant changes unexpectedly.